docs/03-dev-howtos/01-ab-testing.md

```diff
@@ -265,3 +267,85 @@ Optimizely is relatively expensive - several thousand pounds p/month.
 Optimizely is a client-side framework, which is limited for some types of testing.
 
 It adds a rather large overhead to the cookie (mine is 2.5kb).
+
+# Write a Server-Side Test 
+
+If you want to set up your AB test with scala instead of javascript, almost all the steps are different. 
+
+*Advantages:* 
+- Allows you to run tests that just won't work in javascript ie if you want to render a whole different page
+- Avoid javascript rendering problems (eg a jump if a DOM element is rendered and then hidden)
+
+*Disadvantages:* 
+- Can only do AB tests, not multivariant ABC test
+- Only one test can use a given bucket at a time eg. if someone is running a 50:50 test, you can't run a 50:50 until they finish (current setup) 
+- Slightly harder to set up, preview and read test results 
+
+## Configure the test
+
+1. In `Experiments.scala`, create a new object for your test filling in author, and end date as per the other tests. 
+
+2. Ensure that the object name and the name property are the same - but formatted correctly.  
+eg: object `AudioChangeImagePosition` has the name property `audio-change-image-position`.
+```
+object AudioChangeImagePosition extends Experiment(
+  name = "audio-change-image-position",
+  description = "Test the position of the image on audio pages",
+  owners = Owner.group(SwitchGroup.Journalism),
+  sellByDate = new LocalDate(2018, 8, 3),
+  participationGroup = Perc50
+)
+
+```
+3. Add the experiment to the object property `ActiveExperiments.allExperiments` at the top of the file 
+
+4. Switch the test on for your local environment: http://localhost:9000/dev/switchboard You should be able to find it under `Serverside Tests`
+NB: You should be running in `project dev-build` because you will need it to access the /switchboard screen
+
+5. Incorporate the test in your code as a switch, eg:
+
+```
+@import experiments.{ActiveExperiments, AudioChangeImagePosition}
+
+if(ActiveExperiments.isParticipating(AudioChangeImagePosition)) { 
+    //do variant thing 
+ }
+
+```
+
+NB: If your test suddenly stops working in local, check the switchboard again and make sure your test is still switched on. 
```

Comment:
This is probably mentioned elsewhere, but you can set up your own independent switchboard by creating a file `~/.gu/frontend.conf` and adding an entry to it like:
```
devOverrides {
    switches.key=DEV/config/switches-<name>.properties
}
```
This will avoid having your switch states overridden by other developers.
(I'm not sure whether you'll also have to create the file in S3 as well, or if it gets auto-generated.)

Reply:
I will add a note to this effect